miden-tx/src/host/tx_authenticator.rs

```diff
@@ -39,15 +42,53 @@ pub trait TransactionAuthenticator {
     ) -> Result<Vec<Felt>, AuthenticationError>;
 }
 
-// BASIC AUTHENTICATOR
+// AUTH SECRET KEY
 // ================================================================================================
 
 /// Types of secret keys used for signing messages
 #[derive(Clone, Debug)]
+#[repr(u8)]
 pub enum AuthSecretKey {
-    RpoFalcon512(rpo_falcon512::SecretKey),
+    RpoFalcon512(rpo_falcon512::SecretKey) = 0,
+}
+
+impl AuthSecretKey {
+    /// Identifier for the type of authentication key
+    pub fn key_id(&self) -> u8 {
+        match self {
+            AuthSecretKey::RpoFalcon512(_) => 0u8,
+        }
+    }
+}
+
+impl Serializable for AuthSecretKey {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u8(self.key_id());
+        match self {
+            AuthSecretKey::RpoFalcon512(secret_key) => {
+                target.write_bytes(&secret_key.to_bytes());
```

Comment:
Can we not avoid using `to_bytes()` here? (`to_bytes()` would allocate a vector).